File: python_ruby/calculator.py
class Cal(object):
    def __init__(self, v1, v2): #관습적으로 self라는 이름을 씀
        if isinstance(v1, int):
            self.v1 = v1
        else:
            print("parameter1 isn't Integer.")
        if isinstance(v2, int):
            self.v2 = v2
        else:
            print("parameter2 isn't Integer")
    def add(self):
        return self.v1 + self.v2
    def subtract(self):
        return self.v1 - self.v2
    def setV1(self, v):
        if isinstance(v, int):
            self.v1 = v
    def getV1(self):
        return self.v1
c1 = Cal(10, 10)
print(c1.add())
print(c1.subtract())

c2 = Cal(30, 20)
print(c2.add())
print(c2.subtract())
print(Cal(50,50).add())
c3 = Cal("one", 55)
c2.setV1("one")
print(c2.getV1())

File: python_ruby/calculator.rb
class Cal
    def initialize(v1, v2)
        p v1, v2
        @v1 = v1
        @v2 = v2
    end
    def add()
        return @v1 + @v2
    end
    def subtract()
        return @v1 - @v2
    end
    def setV1(v)
        if v.is_a?(Integer)
            @v1 = v
        end
    end
    def getV1()
        return @v1
    end
end

c1 = Cal.new(10, 10)
p c1.add()
p c1.subtract()

c2 = Cal.new(30, 20)
p c2.add()
p c2.subtract()

p Cal.new(50,50).add()

p "below result isn't use object"

def add(v1, v2)
    return v1 + v2
end
def subtract(v1, v2)
    return v1 - v2
end

num1 = 10
num2 = 10
p add(num1, num2)
p subtract(num1, num2)

num3 = 30
num4 = 20
p add(num3, num4)
p subtract(num3, num4)